keyring/backends/OS_X.py

```diff
@@ -41,17 +44,40 @@ def set_password(self, service, username, password):
             raise PasswordSetError("Can't store password on keychain: " "{}".format(e))
 
     def get_password(self, service, username):
-        if username is None:
-            username = ''
+        creds={}
+        if (not username):
+            prompt = "find-generic-password"
+            
+            if (service.find(".com")>=0):
```

Comment:
This logic looks like a copy/paste of line 68 below. Probably this functionality should be its own function or method with a docstring explaining the purpose.